core/crypto/poseidon_hash_test.go

```diff
@@ -0,0 +1,21 @@
+package crypto_test
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/NethermindEth/juno/core/crypto"
+	"github.com/NethermindEth/juno/core/felt"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestPoseidon(t *testing.T) {
+	msg := []*felt.Felt{
+		new(felt.Felt).SetUint64(0),
+		new(felt.Felt).SetUint64(1),
+		new(felt.Felt).SetUint64(2),
+		new(felt.Felt).SetUint64(3),
+	}
+	assert.Equal(t, strings.ToLower("30ACF0EF3C4549E3684B19652015FD9EADF7BDBA2A1A46CB29B9E18E6622296"),
+		crypto.Poseidon(msg).Text(16))
+}
```

Comment:
Please add benchmarks similar to Pedersen hash

Reply:
all done